ref-exchange/src/token_receiver.rs

```diff
@@ -91,11 +104,9 @@ impl FungibleTokenReceiver for Contract {
             self.internal_deposit(sender_id.as_ref(), &token_in, amount.into());
             PromiseOrValue::Value(U128(0))
         } else {
-            // [AUDIT14] shutdown instant swap from interface
-            env::panic(b"Instant Swap Feature Not Open Yet");
-
+            // instant swap
             let message =
-                serde_json::from_str::<TokenReceiverMessage>(&msg).expect("ERR_MSG_WRONG_FORMAT");
+                serde_json::from_str::<TokenReceiverMessage>(&msg).expect(ERR28_WRONG_MSG_FORMAT);
             match message {
                 TokenReceiverMessage::Execute {
                     referral_id,
```

Comment:
I realized a huge security issue with doing it this way. A malicious token contract can provide any "sender_id" account ID and execute given actions on behalf of this account.

It means we have to verify that the given actions are only affect the given token_id.